kornia/filters/sobel.py

```diff
@@ -28,16 +37,29 @@ class SpatialGradient(nn.Module):
         >>> output = kornia.filters.SpatialGradient()(input)  # 1x3x2x4x4
     """
 
-    def __init__(self) -> None:
+    def __init__(self, mode: str = 'Sobel') -> None:
         super(SpatialGradient, self).__init__()
-        self.kernel: torch.Tensor = self.get_sobel_kernel()
+        if mode not in ['Sobel', 'NoBlur']:
+            raise TypeError("Mode should be either type is Sobel\
+                             or NoBlur. Got {}".format(mode))
+        self.mode = mode
+        if self.mode == 'Sobel':
```

Comment:
I don't like if else statements here, maybe we could have under filters.kernels a dict that returns the kernel given its key id